src/runtime/components/server/runtime_support_server.cpp

```diff
@@ -1089,7 +1089,7 @@ namespace hpx { namespace components { namespace server
                     &still_unregistered_options);
 
                 std::string still_unknown_commandline;
-                for (std::size_t i = 1; i != still_unregistered_options.size();
+                for (std::size_t i = 1; i < still_unregistered_options.size();
```

Comment:
@hkaiser would you happen to know the significance of starting at `i = 1`? It's not really clear from the context.

Reply:
@msimberg: honestly, I don't remember. I'll try looking at it at runtime later.

Reply:
@msimberg the first element of that vector is the equivalent to `argv[0]`, i.e. it refers to the path of the launched executable.

Reply:
It seems like that's the case when `allow_unknown=1`, but when it's `0` the vector can be empty. Whether that's a bug or not I can't tell.

Reply:
@msimberg that could be the case and your fix addresses this issue.